stl/inc/xstring

```diff
@@ -217,6 +220,9 @@ public:
     using pos_type   = streampos;
     using off_type   = streamoff;
     using state_type = mbstate_t;
+#if _HAS_CXX20
+    using comparison_category = strong_ordering;

```

Comment:
This was removed in my last commit of my previous PR. Not sure if there is a reason to add it again.

Reply:
Nice catch! This is intentional. Casey's comment https://github.com/microsoft/STL/pull/1635#discussion_r577102048 was actually incorrect, which is why https://github.com/microsoft/STL/pull/1635/commits/10886da7e2dcd34d042afe2fde6fb0442a99287d caused the tests to fail in https://github.com/microsoft/STL/runs/1923245294 with:
```
C:\a\1\s\tests\std\tests\P1614R2_spaceship\test.cpp(451): error C2607: static assertion failed
```
I fixed this in https://github.com/microsoft/STL/pull/1635/commits/7a79abbc0e7258a65a61f167576407af581fa428 before merging.

Although `_Char_traits` defines `comparison_category`:
https://github.com/microsoft/STL/blob/5547f0f3389999dcd732d3316d621386df5a0531/stl/inc/xstring#L35-L44

`_WChar_traits` derives from `_Char_traits` *privately*, so the base class typedef is inaccessible to the outside world. Therefore we need to provide an identically-named typedef in the `public` section of the derived class:
https://github.com/microsoft/STL/blob/5547f0f3389999dcd732d3316d621386df5a0531/stl/inc/xstring#L211-L225

Similarly for `_Narrow_char_traits`:
https://github.com/microsoft/STL/blob/5547f0f3389999dcd732d3316d621386df5a0531/stl/inc/xstring#L352-L366

Note that this is not a redefinition - a derived class is allowed to have a typedef of the same name as a base class, and name lookup will find the derived one's first.

This allows the tests to succeed, and I have observed no `/permissive-` issues. (An earlier version of the typedefs that you tried could have caused `/permissive-` issues depending on the exact code; I can't guess what happened without seeing what failed.)